Fix member variable order to build with -Wreorder

I'm trying to build Pyne using MOOSE's environment, which passes the `-Wreorder` compile flag. This is hitting the following error:

```
/home/anovak/cardinal/contrib/DAGMC/src/pyne/pyne.h: In constructor 'pyne::Sampler::Sampler(std::string, std::string, std::vector<double>, std::string)':
/home/anovak/cardinal/contrib/DAGMC/src/pyne/pyne.h:6243:25: error: 'pyne::Sampler::e_bounds' will be initialized after [-Werror=reorder]
 6243 |     std::vector<double> e_bounds;  ///< Energy boundaries
      |
```

Solution is to just fix the member variable order in the header file.

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -9,6 +9,7 @@ Next Version
    * remove setup-QEMU step from composite action(#1461)
    * add composite actions to github workflows(#1459)
    * fixing typos in user warning message (#1456)
+   * fix reorder error in member variables (#1466)
 
 v0.7.7
 ======

--- a/src/source_sampling.h
+++ b/src/source_sampling.h
@@ -209,12 +209,12 @@ namespace pyne {
     // problem parameters
     std::string filename; ///< MOAB mesh file path
     std::string src_tag_name; ///< Unbiased source density distribution
-    std::string bias_tag_name; ///< Biased source density distribution
     std::string cell_number_tag_name; ///< Cell number tag
     std::string cell_fracs_tag_name; ///< Cell volume fraction tag
     std::map<std::string, std::string> tag_names; /// < tag names
     std::string e_bounds_tag_name; ///< Energy boundaries tag name
     std::vector<double> e_bounds;  ///< Energy boundaries
+    std::string bias_tag_name; ///< Biased source density distribution
     int num_e_groups; ///< Number of groups in tag \a _src_tag_name
     int num_bias_groups; ///< Number of groups tag \a _bias_tag_name
     int max_num_cells; /// Max number of cells in voxels

